datasets.py

```diff
@@ -39,6 +41,8 @@ def read_transcript(data_path):
 		self.examples = [list(g) for data_path in data_paths for k, g in itertools.groupby(sorted(read_transcript(data_path), key = transcripts.sort_key), key = transcripts.group_key)]
 		self.examples = list(sorted(self.examples, key = duration))
 		self.examples = list(filter(lambda example: (min_duration is None or min_duration <= duration(example)) and (max_duration is None or duration(example) <= max_duration), self.examples))
+		self.examples = [e for e in self.examples if transcripts.audio_name(e[0]) not in set(exclude)]
```

Comment:
Conversion to a set at every iteration is costly. Please replace it with just `not in exclude`. Let's put the burden of providing a set as exclude arg on the caller. Are there any usages at the moment?